dashboard/lib/projects_list.rb

```diff
@@ -116,6 +117,41 @@ def fetch_featured_published_projects
       return featured_published_projects
     end
 
+    def fetch_section_libraries(section)
+      project_types = PUBLISHED_PROJECT_TYPE_GROUPS[:library]
+      section_students = section.students
+      [].tap do |projects_list_data|
+        student_storage_ids = PEGASUS_DB[:user_storage_ids].
+          where(user_id: section_students.pluck(:id)).
+          select_hash(:user_id, :id)
+        section_students.each do |student|
+          next unless student_storage_id = student_storage_ids[student.id]
+          PEGASUS_DB[:storage_apps].
+            where(storage_id: student_storage_id, state: 'active').
+            where(project_type: project_types).
+            where("value->'$.libraryName' IS NOT NULL").
+            each do |project|
+              # The channel id stored in the project's value field may not be reliable
+              # when apps are remixed, so recompute the channel id.
+              channel_id = storage_encrypt_channel_id(student_storage_id, project[:id])
+              project_data = get_library_row_data(project, channel_id, student)
+              projects_list_data << project_data if project_data
```

Comment:
Note: Should filter this to unique projects

Reply:
no need to filter out dups if you change the query as described in the previous comment